### PR TITLE
add link/token/create support

### PR DESCRIPTION
**Pull request by otherchen**
- add support for /link/token/create endpoint

### CODE REVIEW

File: lib/plaid/products/link_token.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module Plaid
+  # Public: Class used to call the LinkToken sub-product.
+  class LinkToken < BaseProduct
+    def create(configs)
+      post_with_auth 'link/token/create',
+                     CreateResponse,
```
Comment (davidzhanghp) [inline lib/plaid/products/link_token.rb:6]: please for your sake let's make an array with list of keys and look over it to create the object

File: lib/plaid/products/link_token.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+$link_token_field_names = [
```
Comment (davidzhanghp) [inline lib/plaid/products/link_token.rb:1]: nit let's have this as

```
LINK_TOKEN_FIELD_NAMES = %i[
Attach files by dragging & dropping, selecting or pasting them.
   user
   client_name
   products
   country_codes
   language
   webhook
   access_token
   link_customization_name
   redirect_uri
   android_package_name
   account_filters
   cross_app_item_add
   payment_initiatio
]
```

File: lib/plaid/products/link_token.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+   :payment_initiation
+]
+
+module Plaid
+  # Public: Class used to call the LinkToken sub-product.
+  class LinkToken < BaseProduct
+    def create(configs)
+      body = {}
+
+      for field in $link_token_field_names do
```
Comment (davidzhanghp) [inline lib/plaid/products/link_token.rb:23]: nit more idiomati cruby would be

```
LINK_TOKEN_FIELD_NAMES.each do |field|
  body[field] = configs[field]
end
```

File: test/test_link_token.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+# Internal: The test for Plaid::Item.
+class PlaidLinkTokenTest < PlaidTest
+  def test_create_required
+    link_token_response = client.link_token.create(
+      user: { client_user_id: '123-fake-user-id' },
+      client_name: 'Plaid Test',
+      products: %w[auth transactions]
+    )
+    refute_empty(link_token_response.token)
```
Comment (davidzhanghp) [inline test/test_link_token.rb:11]: link_token

File: lib/plaid/products/link_token.rb (hunk 2 above)
Comment (otherchen) [reply lib/plaid/products/link_token.rb:1]: ⚡

File: lib/plaid/products/link_token.rb (hunk 3 above)
Comment (otherchen) [reply lib/plaid/products/link_token.rb:23]: ⚡

**Pull request merged**